src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java

```diff
@@ -104,75 +105,59 @@ public void onCommandEvent(final CommandEvent event) {
     }
 
     @Override
-    public Optional<CommandMapping> register(Object plugin, CommandCallable callable, String... alias) {
-        return register(plugin, callable, Arrays.asList(alias));
-    }
-
-    @Override
-    public Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases) {
-        return register(plugin, callable, aliases, Functions.<List<String>>identity());
+    public Optional<CommandMapping> register(Object plugin, CommandCallable<Boolean> callable, String... aliases) {
+        return register(plugin, callable, Arrays.asList(aliases), Functions.<List<String>>identity());
     }
 
     @Override
-    public Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases,
+    public Optional<CommandMapping> register(Object plugin, CommandCallable<Boolean> callable, List<String> aliases,
             Function<List<String>, List<String>> callback) {
         checkNotNull(plugin);
-
-        Optional<PluginContainer> containerOptional = this.pluginManager.fromInstance(plugin);
```

Comment:
@simon816 Sorry about that, it was just really annoying with the merge conflict.